DisruptorUnity3d/Assets/RingBuffer.cs

```diff
@@ -81,19 +81,19 @@ public bool TryDequeue(out T obj)
         /// <param name="item"></param>
         public void Enqueue(T item)
         {
-            var next = _producerCursor.ReadFullFence() + 1;
+            var next = _producerCursor.ReadAcquireFence() + 1;
 
             long wrapPoint = next - _entries.Length;
-            long min = _consumerCursor.ReadFullFence();
+            long min = _consumerCursor.ReadAcquireFence(); 
 
             while (wrapPoint > min)
             {
-                min = _consumerCursor.ReadFullFence();
+                min = _consumerCursor.ReadAcquireFence();
                 Thread.SpinWait(1);
             }
 
             this[next] = item;
-            _producerCursor.WriteUnfenced(next);
+            _producerCursor.WriteReleaseFence(next); // makes sure we write the data in _entries before we update the producer cursor
```

Comment:
The problem is here. A write-release must occur to make sure that the write to the array occurs before the write to the cursor.